### PR TITLE
elan-init, lean: depend on coreutils

### PR DESCRIPTION
**Pull request by Julian**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

`leanpkg` is a thing that [depends on `coreutils`](https://github.com/leanprover-community/lean/blob/a528e9be5890604872b8036b529ab919d7531837/bin/leanpkg#L6-L11).

`lean` therefore, which installs it, depends on `coreutils`, as does `elan-init`, which installs `lean` and transitively therefore does as well.

### CODE REVIEW

File: Formula/lean.rb [hunk 1]
```diff
@@ -45,6 +46,7 @@ def id' {α : Type} (x : α) : α := x
           split, repeat { assumption }
       end
     EOS
-    system "#{bin}/lean", testpath/"hello.lean"
+    system bin/"lean", testpath/"hello.lean"
+    system bin/"leanpkg", "--help"
```
Comment (carlocab) [inline Formula/lean.rb:50]:
```suggestion
    system bin/"leanpkg", "help"
```
Looks like passing `--help` is unrecognised, and returns a non-zero exit code.

File: Formula/elan-init.rb [hunk 2]
```diff
@@ -49,5 +50,6 @@ def id' {α : Type} (x : α) : α := x
       end
     EOS
     system bin/"lean", testpath/"hello.lean"
+    system bin/"leanpkg", "--help"
```
Comment (carlocab) [inline Formula/elan-init.rb:53]:
```suggestion
    system bin/"leanpkg", "help"
```

File: Formula/lean.rb (hunk 1 above)
Comment (Julian) [reply Formula/lean.rb]: Was just fixing that, sorry, you mean squash the 2 formulae changes? Or just squash the fixup I just did? If so think I did so, lemme know if you meant something else.

**Comment by carlocab** [comment]
Oops, another thing we missed. Both `lean` and `elan-init` need appropriate `conflicts_with` statements. For example:
```ruby
conflicts_with "lean", because: "`lean` and `elan-init` install the same binaries"
```
or similar. This should resolve the CI failure on the Intel nodes.

**Comment by Julian** [comment]
A ha, cool, will add that too thanks!

**Review by carlocab — APPROVED** [review]
Thanks, @Julian.

Merging the old-fashioned way since this doesn't need new bottles.

**Pull request merged**